python/GafferSceneUI/EditScopeUI.py

```diff
@@ -101,3 +108,51 @@ def __pruningKeyPress( viewer, event ) :
 			GafferScene.EditScopeAlgo.setPruned( editScope, selection, True )
 
 	return True
+
+def __visibilityKeyPress( viewer, event ) :
+
+	if not ( event.key == "H" and event.Modifiers.Control ) :
+		return False
+
+	if not isinstance( viewer.view(), GafferSceneUI.SceneView ) :
+		return False
+
+	editScope = viewer.view().editScope()
+	if editScope is None or Gaffer.MetadataAlgo.readOnly( editScope ) :
+		return True
+
+	sceneGadget = viewer.view().viewportGadget().getPrimaryChild()
+	selection = sceneGadget.getSelection()
+
+	if selection.isEmpty() :
+		return True
+
+	inspector = GafferSceneUI.Private.AttributeInspector(
+		viewer.view()["in"].getInput(),
+		viewer.view()["editScope"],
+		"scene:visible"
+	)
+
+	with viewer.getContext() as context :
+		if (
+			not editScope["enabled"].getValue() or
+			( "AttributeEdits" in editScope and not editScope["AttributeEdits"]["enabled"].getValue() )
```

Comment:
We shouldn't rely on the name of the processor here - there's nothing stopping a user modifying it. Instead we should use `EditScope.acquireProcessor( "AttributeEdits", createIfNecessary = False )`, which uses metadata instead.